ddsketch/ddsketch.go

```diff
@@ -111,7 +127,24 @@ func (s *DDSketch) MergeWith(other *DDSketch) error {
 	if !s.IndexMapping.Equals(other.IndexMapping) {
 		return errors.New("Cannot merge sketches with different index mappings.")
 	}
-	s.store.MergeWith(other.store)
+	s.positiveValueStore.MergeWith(other.positiveValueStore)
+	s.negativeValueStore.MergeWith(other.negativeValueStore)
 	s.zeroCount += other.zeroCount
 	return nil
 }
+
+func (s *DDSketch) ToProto() *sketchpb.DDSketch {
+	return &sketchpb.DDSketch{
+		Mapping:        s.IndexMapping.ToProto(),
+		PositiveValues: s.positiveValueStore.ToProto(),
+		NegativeValues: s.negativeValueStore.ToProto(),
+		ZeroCount:      s.zeroCount,
+	}
+}
+
+func (s *DDSketch) FromProto(pb *sketchpb.DDSketch) {
```

Comment:
To use `FromProto`, I assume you would have to initialize a `DDSketch` with a function like `MemoryOptimalCollapsingLowestSketch` and then call `FromProto`. If so, how would you get the relative accuracy and max number of bins to initialize the `DDSketch` in the first place?

Also, I tried to test this with

```go

func TestProto(t *testing.T){
	sketch, err := MemoryOptimalCollapsingLowestSketch(0.9, 20)
	require.NoError(t, err)

	for i := 0; i < 500; i++ {
		v := rand.NormFloat64()
		sketch.Accept(v)
	}
	
	sketch2, err := MemoryOptimalCollapsingLowestSketch(0.9, 20)
	require.NoError(t, err)
	sketch2.FromProto(sketch.ToProto())

	qs := []float64{0.5, 0.75, 0.9, 1}
	for _, q := range qs {
		q1, _ := sketch.getValueAtQuantile(q)
		q2, _ := sketch.getValueAtQuantile(q)

		// This works
		assert.Equal(t, q1, q2)
	}

	// This fails (sketch2 has lots of empty bins)
	assert.Equal(t, sketch, sketch2)
}
```

And the last assertion fails since `sketch2` has a store with 257 bins, most of them empty. Not sure if that's to be expected but I thought it should have been 20 so I wanted to bring it up just in case.

Reply:
Yes, thanks for catching this! I hadn't added the `FromProto` method to the `CollapsingHighestDenseStore`. I also added this test.

Reply:
I am glad it helped :)

Reply:
The relative accuracy will be set to that of the serialized sketch, but the maxNumBins of the store of the calling sketch will be retained, so you can call `FromProto` from a sketch with a larger or smaller store than the serialized sketch.

Reply:
👍, thanks for clarifying! If the relative accuracy gets overridden it would be less confusing to me to change `FromProto` from a method to a function like
 
```go
func FromProto(maxNumBins int, pb *sketchpb.DDSketch) *DDSketch
```

but I don't have a lot of context on all the possible use cases so probably I am missing something.